src/backend/api/Fusion.Resources.Api/Controllers/Requests/RequestsController.cs

```diff
@@ -321,8 +312,103 @@ public async Task<ActionResult<ApiContractPersonnelRequest>> ProvisionContractor
             }
         }
 
+        #region Comments
+
+        [HttpPost("/projects/{projectIdentifier}/contracts/{contractIdentifier}/resources/requests/{requestId}/comments")]
+        public async Task<ActionResult> AddRequestComment([FromRoute]ProjectIdentifier projectIdentifier, Guid contractIdentifier, Guid requestId, [FromBody] RequestCommentRequest create)
+        {
+            #region Authorization
+
+            var authResult = await Request.RequireAuthorizationAsync(r =>
+            {
+                r.AlwaysAccessWhen().FullControl();
+
+                r.AnyOf(or =>
+                {
+                    or.ContractAccess(ContractRole.Any, projectIdentifier, contractIdentifier);
+                    or.BeContractorInContract(contractIdentifier);
+                });
+            });
+
+            if (authResult.Unauthorized)
+                return authResult.CreateForbiddenResponse();
+
+            #endregion
+
+            await DispatchAsync(new AddComment(requestId, create.Content));
+
+            return NoContent();
+        }
+
+        [HttpPut("/projects/{projectIdentifier}/contracts/{contractIdentifier}/resources/requests/{requestId}/comments/{commentId}")]
+        public async Task<ActionResult> UpdateRequestComment(
+            [FromRoute]ProjectIdentifier projectIdentifier, Guid contractIdentifier, Guid requestId, Guid commentId, [FromBody] RequestCommentRequest update)
+        {
+            var comment = await DispatchAsync(new GetRequestComment(commentId));
+
+            if (comment == null)
+                return FusionApiError.NotFound(commentId, "Comment not found");
+
+            #region Authorization
+
+            var authResult = await Request.RequireAuthorizationAsync(r =>
+            {
+                r.AlwaysAccessWhen().FullControl();
+                r.AnyOf(r => r.BeCommentAuthor(comment));
```

Comment:
This would cause that the user must be the comment author AND one of the requirements below. They should all be in the same block.

Reply:
If I put them in the same block, would that not allow users to edit other users' comments?